### ISSUE TITLE
Do queries affect annotations?

### ISSUE THREAD

**Issue by mohona7428** [microsoft/restler-fuzzer#768]
### Description

I'm trying to fuzz an API where the POST request must be completed first with some parameters like name, category, etc. 
Then a GET request must be performed using those parameters as queries. 
All the other requests (PUT, DELETE, etc) depend on getting the ID from the GET request.

1. I tried to write an annotation for getting the ID from the GET request as something like this:
```
{
	    "producer_endpoint": "/setup",

            "producer_method": "GET",

            "producer_resource_name": "id",

            "consumer_param": "setID"
}
```

But there's no change in the grammar file after compiling. I've enabled GET producers in the config file as well.

2. Is it possible to add the dependency of the GET request getting the query parameters from the POST? 
The parameters are user generated and only present in the request body.

**Comment by marina-p** [comment]
Hello @mohona,

I believe everything you are asking for should be possible to configure.  Just to make sure I understand the API pattern:

POST /product {name = "Name"; category = "Food"} 
-> returns 200 OK?  or 201 with Location etc. ?

GET /product?name=Name&category=Food
-> returns 200 OK {"id" = 12345; name = "Name", category = "Food"...}

PUT /product/12345  <-- /product/{setID} is the path as declared in OpenAPI

Can you please confirm this is the setup?  I believe the easiest way to troubleshoot would be to have an OpenAPI spec sample to reproduce the issue.   I am attaching an example with a dependency both as in your (1) and (2). 

Config:
```
{
  "SwaggerSpecFilePath": [
    "../swagger.json"
  ],
  "DataFuzzing": true,
  "AnnotationFilePath": "annotations.json"
}
```

Annotation file:
```
{
  "x-restler-global-annotations": [
    {
      "producer_endpoint": "/setup",
      "producer_method": "GET",
      "producer_resource_name": "id",
      "consumer_param": "setID"
    },
    {
      "producer_endpoint": "/setup",
      "producer_method": "POST",
      "producer_resource_name": "/name",
      "consumer_param": "product_name",
      "consumer_endpoint": "/setup",
      "consumer_method": "GET"
    }

  ]
}
```

Swagger and resulting grammar with generated dependencies attached.

If there is something different about your Swagger/OpenAPI spec that makes the annotations not take effect, please send me your version of the spec that reproduces the issue.  Or, please let me know what fixes you applied to make i
[grammar.py.txt](https://github.com/microsoft/restler-fuzzer/files/11638115/grammar.py.txt)
t work (perhaps there is a doc improvement we can make to better describe such examples).
[swagger.json.txt](https://github.com/microsoft/restler-fuzzer/files/11638113/swagger.json.txt)


Thanks,

Marina

**Comment by mohona7428** [comment]
Hi, these annotations really helped. I couldn't find any information in the documentation about how to retrieve a parameter from the body (eg. '/name').

However, I've run into another issue: one of the parameters needs to be sent as a string in the GET query, but it is in the form of an array in the POST request.

For example:
POST /product {name = "name", models = ["abcd", "efgh"]}

GET /product?name="name"&model="abcd"

Is there a way to configure restler to extract the values from within the array?

**Comment by mohona7428** [comment]
I tried to workaround this by manually changing the grammar to include the brackets and the request is giving a 200 status code however it's rendered invalid due to a parser failure.
